File: src/main/java/com/weeth/domain/account/application/mapper/AccountMapper.java
package com.weeth.domain.account.application.mapper;

import com.weeth.domain.account.application.dto.AccountDTO;
import com.weeth.domain.account.application.dto.ReceiptDTO;
import com.weeth.domain.account.domain.entity.Account;
import org.mapstruct.*;

import java.util.List;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface AccountMapper {

    @Mapping(target = "accountId", source = "account.id")
    @Mapping(target = "receipts", source = "receipts")
    @Mapping(target = "time", source = "account.modifiedAt")
    AccountDTO.Response to(Account account, List<ReceiptDTO.Response> receipts);

    @Mapping(target = "currentAmount", source = "totalAmount")
    Account from(AccountDTO.Save dto);
}

File: src/main/kotlin/com/weeth/domain/account/application/mapper/AccountMapper.kt
package com.weeth.domain.account.application.mapper

import com.weeth.domain.account.application.dto.response.AccountResponse
import com.weeth.domain.account.application.dto.response.ReceiptResponse
import com.weeth.domain.account.domain.entity.Account
import org.springframework.stereotype.Component

@Component
class AccountMapper {
    fun toResponse(
        account: Account,
        receipts: List<ReceiptResponse>,
    ): AccountResponse =
        AccountResponse(
            accountId = account.id,
            description = account.description,
            totalAmount = account.totalAmount,
            currentAmount = account.currentAmount,
            time = account.modifiedAt,
            cardinal = account.cardinal,
            receipts = receipts,
        )
}
